### ISSUE TITLE
`download.MERRA()` doesn't fully respect overwrite arg

### ISSUE THREAD

**Issue by Aariq** [PecanProject/pecan#2998]
### Bug Description

When `do_conversions(settings, overwrite.met = FALSE)` is run, files get overwritten.  The `overwrite.met` arg isn't used to determine whether to write a .nc file in the following lines:
https://github.com/PecanProject/pecan/blob/01b370430304dfd965aa7b31e15f209ecc78775e/modules/data.atmosphere/R/download.MERRA.R#L103-L111

Judging by the speed of the reprex below, the downloading isn't repeated (although a message indicating the download is happening is still printed)

### To Reproduce

``` r
library(PEcAn.data.atmosphere)
#> Warning: package 'PEcAn.data.atmosphere' was built under R version 4.2.1
outdir <- tempdir()
start_date <- "2009-06-01"
end_date <- "2009-06-02"
dat <- 
  download.MERRA(outdir, start_date, end_date,
                 lat.in = 45.3, lon.in = -85.3, overwrite = FALSE)
#> 2022-08-01 17:36:03 DEBUG  [download.MERRA] : 
#>    Downloading 2009-06-01 (1 of 2) 
#> 2022-08-01 17:36:04 DEBUG  [download.MERRA] : 
#>    Downloading 2009-06-02 (2 of 2) 

dat <- 
  download.MERRA(outdir, start_date, end_date,
                 lat.in = 45.3, lon.in = -85.3, overwrite = FALSE)
#> 2022-08-01 17:36:51 DEBUG  [download.MERRA] : 
#>    Downloading 2009-06-01 (1 of 2) 
#> 2022-08-01 17:36:51 DEBUG  [download.MERRA] : 
#>    Downloading 2009-06-02 (2 of 2) 
#> 2022-08-01 17:36:51 WARN   [download.MERRA] : 
#>    Target file 
#>    /var/folders/wr/by_lst2d2fngf67mknmgf4340000gn/T//RtmpOn2aNs/file2da42ca43976/MERRA.2009.nc 
#>    already exists. It will be overwritten. 
```

<sup>Created on 2022-08-01 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

## Expected behavior

If `overwrite = FALSE` and the files exist, a different message should be printed and no attempt to write the file should be made.

**Comment by Aariq** [comment]
Related to #1922 and possibly #1571

**Comment by meetagrawal09** [comment]
@Aariq I think this issue can be closed since it is resolved by #3169.
